pygments/lexers/arturo.py

```diff
@@ -0,0 +1,312 @@
+import re
+
+from pygments.lexer import bygroups, default, DelegatingLexer, do_insertions,\
+    include, RegexLexer, this, using, words
+from pygments.token import Comment, Generic, Keyword, Name, Number, Operator,\
+    Other, Punctuation, String, Text
+
+from pygments.util import ClassNotFound, get_bool_opt
+
+class ArturoLexer(RegexLexer):
+
+    name = 'Arturo'
+    aliases = ['Arturo', 'Art', 'arturo', 'art']
+    filenames = ['*.art']
+
+    def handle_annotated_strings(self, match):
+        """Adds syntax from another languages inside annotated strings
+
+        match args:
+            1:open_string,
+            2:exclamation_mark,
+            3:lang_name,
+            4:space_or_newline,
+            5:code,
+            6:close_string
+        """
+        from pygments.lexers import get_lexer_by_name
+
+        # Header's section
+        yield match.start(1),  String.Double  ,  match.group(1)
+        yield match.start(2),  String.Interpol,  match.group(2)
+        yield match.start(3),  String.Interpol,  match.group(3)
+        yield match.start(4),  Text.Whitespace,  match.group(4)
+
+        lexer = None
+        if self.handle_annotateds:
+            try:
+                lexer = get_lexer_by_name( match.group(3).strip() )
+            except ClassNotFound:
+                pass
+        code = match.group(5)
+
+        if lexer is None:
+            yield match.group(5), String, code
+        else:
+            yield from do_insertions([], lexer.get_tokens_unprocessed(code))
+
+        yield match.start(6), String.Double, match.group(6)
+
+
+
+    tokens = {
+        'root': [
+            include('comments'),
+            include('constants'),
+            include('builtin_functions'),
+            include('operators'),
+            (r'.', Name),
+        ],
+
+        'comments': [
+            (r'(;)(.*)$', Comment.Single),
+            include('shebang'),
+        ],
+            'shebang': [
+                (r'^((\s#!)|(#!))(.*)$',
+                        Comment.Hashbang)
+            ],
+
+        'operators': [
+            include('punctuation'),
+            include('sugar'),
+
+            (r'<:|:>|:<|:>'
```

Comment:
better use `words()` for such lists of literals, it is easier to read and avoids common mistakes - for example, this will never match `!!` since the `!` alternate comes first.

Reply:
In this case, do I should escape special characters as `?`?

Reply:
- [x] [replace with words function](https://github.com/pygments/pygments/pull/2259/commits/485b840311819421b1744f630f9490baaa53dda6)